src/openrct2/peep/Peep.cpp

```diff
@@ -765,7 +765,8 @@ bool Peep::Place(TileCoordsXYZ location, bool apply)
     TileElement* tileElement = reinterpret_cast<TileElement*>(pathElement);
     if (!pathElement)
     {
-        tileElement = reinterpret_cast<TileElement*>(map_get_surface_element_at(location.x, location.y));
+        tileElement = reinterpret_cast<TileElement*>(
```

Comment:
`tileElement = reinterpret_cast<TileElement*>(map_get_surface_element_at(location.ToCoordsXYZ));` should be sufficient.